src/v/storage/ntp_config.h

```diff
@@ -290,6 +294,47 @@ class ntp_config {
                           : cluster_default;
     }
 
+    std::optional<std::chrono::milliseconds> tombstone_retention_ms() const {
+        if (is_read_replica_mode_enabled()) {
+            // RRR sanity check.
+            return std::nullopt;
+        }
+        /*TODO(willem):
```

Comment:
These TODOs are resolved in this PR in a later commit (after the cluster property is added).